Automattic/wp-calypso
# Add some load optimisations and an error screen for loading errors

#### Changes proposed in this Pull Request

* Adds an error screen for when an error occurs during loads
* Improves the underlying query for fetching mailboxes by adding reasonable automatic retrials

#### Testing instructions


Note: You need a sandbox to simulate a backend error.

* Check out this branch
* Sandbox the public-api
* Load the inbox page and select a site with existing mailboxes
* Confirm that you see your list of mailboxes
* Modify the backend REST endpoint `/emails/mailboxes/` to return an HTTP error i.e. response code other 200
* Confirm that the Calypso URL `/inbox` shows an error view:



<img width="500" alt="Screenshot 2021-09-24 at 9 14 42 PM" src="https://user-images.githubusercontent.com/277661/134734598-8e098bf2-e4e4-47ac-b5e8-387aad374dca.png">
